File: 8-string-to-integer-atoi/8-string-to-integer-atoi.py
class Solution:
    def myAtoi(self, s: str) -> int:
        sign=1
        s=s.strip()
        if s=="":
            return 0
        if s[0]=='-':
            s=s[1:]
            sign*=-1
        elif s[0]=='+':
            s=s[1:]
        mx = 2**31
        mn=-mx
        mx-=1
        num=0
        for i in s:
            if not i.isnumeric():
                break
            num=num*10 + int(i)
            if sign*num<=mn:
                num=abs(mn)
                break
            if sign*num>=mx:
                num=abs(mx)
                break
        return sign*num

File: 8-string-to-integer-atoi/8-string-to-integer-atoi.cpp
class Solution {
public:
    int myAtoi(string s) {
        long x = 0;
        int curr = 0;
        while(curr<s.size() && s[curr]==' ') curr++;
        bool neg = false;
        if(s[curr]=='+' || s[curr]=='-'){
            if(s[curr]=='-') neg=true;
            curr++;
        }
        while(curr<s.size() && s[curr]>='0' && s[curr]<='9'){
            x*=10;
            x+=s[curr]-'0';
            curr++;
            if(x>INT32_MAX){
                x = INT32_MAX;
                x+=2;
                break;
            }
        }
        if(neg){
            x*=-1;
            if(x<INT32_MIN) x = INT32_MIN;
        }
        else if(x>INT32_MAX){
            x = INT32_MAX;
        }
        return x;
    }
};